cli/azd/internal/appdetect/docker_test.go

```diff
@@ -0,0 +1,25 @@
+package appdetect
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestParsePorts(t *testing.T) {
+	tests := []struct {
+		portString    string
+		expectedPorts []Port
+	}{
+		{"", nil},
+		{"80", []Port{{80, "tcp"}}},
+		{"80 3100", []Port{{80, "tcp"}, {3100, "tcp"}}},
+		{"80 3100/udp", []Port{{80, "tcp"}, {3100, "udp"}}},
+		{" 80/tcp 3100/udp ", []Port{{80, "tcp"}, {3100, "udp"}}},
+	}
+	for _, tt := range tests {
+		t.Run(tt.portString, func(t *testing.T) {
+			actual, _ := parsePorts(tt.portString)
```

Comment:
We can also assert that

`assert.NoError(t, err)`

just for extra validation.

Reply:
Updated.